packages/common/src/models/USDCTransactions.ts

```diff
@@ -0,0 +1,79 @@
+import { Nullable } from '../utils/typeUtils'
+
+import { StringUSDC } from './Wallet'
+
+export enum USDCTransactionType {
+  PURCHASE = 'PURCHASE',
+  PURCHASE_CONTENT = 'PURCHASE_CONTENT',
+  TRANSFER = 'TRANSFER'
+}
+
+export enum USDCTransactionMethod {
+  // Transfer methods
+  SEND = 'SENT',
+  RECEIVE = 'RECEIVED',
+
+  // Purchase Methods
+  STRIPE = 'STRIPE'
+}
+
+export enum USDCTransactionMetadataType {
+  PURCHASE_USDC = 'PURCHASE_USDC',
+  PURCHASE_CONTENT = 'PURCHASE_CONTENT',
+  TRANSER = 'TRANSFER'
+}
+
+export type InAppUSDCPurchaseMetadata = {
+  discriminator: USDCTransactionMetadataType.PURCHASE_USDC
+  usd: string
+  usdc: StringUSDC
+  purchaseTransactionId: string
```

Comment:
fyi this could be redundant as the transaction might be able to directly go to the userbank (still probably need it though to associate on identity? I forget)

Reply:
In my mind, I had imagined this was the id of a Solana or Stripe transaction that we wanted to keep track of because we weren't going to index it. But if this is going to be a duplicate of the `signature` column on `usdc_transactions`, then I don't think we need to store it separately.